audit/conn/internal/writer/writer.go

```diff
@@ -0,0 +1,55 @@
+// Package writer implements a writer to a remote audit server of any type that can be connected to
+// by a net.Conn object.
+package writer
+
+import (
+	"bufio"
+	"context"
+	"net"
+	"time"
+
+	"github.com/microsoft/go-otel-audit/audit/msgs"
+)
+
+type now func() time.Time
+
+// Conn is a generic writer to a remote audit server. It can use anything
+// that implements the net.Conn interface.
+type Conn struct {
+	conn net.Conn
+	buff *bufio.Writer
+
+	now now
+}
+
+// New creates a new connection to the remote audit server.
+func New(conn net.Conn) *Conn {
+	return &Conn{conn: conn, buff: bufio.NewWriter(conn), now: time.Now}
+}
+
+// Write writes a message to the remote audit server. Setting a timeout
+// on the context will set the write deadline.
+func (c *Conn) Write(ctx context.Context, msg msgs.Msg) error {
+	if t, ok := ctx.Deadline(); ok {
+		c.conn.SetWriteDeadline(t)
```

Comment:
How does this work with possibly multiple writes across a single stable connection, possibly with some having a deadline and others not? I guess it means that the last context that _has_ a deadline will persist on subsequent `Write` calls? That's a somewhat subtle behavior where the global state of the `Conn` is mutated during a call that a user might not necessarily _expect_.

OTOH, since `Write` doesn't take a `ctx` maybe this is just "the way"?

Reply:
Conns generally don't have multiple writers unless they explicitly say that. Networking wise, you can't have multiple writers calling something like Write(), you either protect via mutex or channel. In this case, upper ends use a channel.

This is why you can have SetWriteDeadline() used in this manner, because it won't persist. If the deadline isn't set, the zero value is returned. That sets the write deadline to infinite.

I included support for these "cancellations" on messages, because .Net has it.  I noted in the design (in the next PR) that I think this is a terrible idea we should get rid of.  My big problem is that the logger inherits the parent context timeout which unintentionally kills the message being sent. I suggest in the documentation that the caller use Context.WithoutCancel(ctx) when passing in.

Reply:
I wasn't really saying in parallel, more in sequence (N writes in sequence each with a different ctx w/ a different deadline, followed by some writes with ctx with no deadline)

Reply:
Got it sorted out offline.